ddtrace/tracer/option_test.go

```diff
@@ -1252,13 +1259,21 @@ func TestWithHeaderTags(t *testing.T) {
 	})
 
 	t.Run("env-override", func(t *testing.T) {
+		defer globalconfig.ClearHeaderTags()
 		assert := assert.New(t)
 		os.Setenv("DD_TRACE_HEADER_TAGS", "unexpected")
 		defer os.Unsetenv("DD_TRACE_HEADER_TAGS")
 		newConfig(WithHeaderTags([]string{"expected"}))
 		assert.Equal(ext.HTTPRequestHeaders+".expected", globalconfig.HeaderTag("Expected"))
 		assert.Equal(1, globalconfig.HeaderTagsLen())
 	})
+
+	// ensures we cleaned up global state
+	t.Run("default-off-at-end", func(t *testing.T) {
+		defer globalconfig.ClearHeaderTags()
+		newConfig()
+		assert.Equal(t, 0, globalconfig.HeaderTagsLen())
```

Comment:
Wouldn't it be better to move this assertion into the main test rather than a subtest? This way it also runs when the `go test -run` flag is used.

Reply:
Great suggestion, thank you! Done.